network/hostport.go

```diff
@@ -236,3 +238,115 @@ func EnsureFirstHostPort(first HostPort, hps []HostPort) []HostPort {
 	result = append([]HostPort{first}, result...)
 	return result
 }
+
+// UniqueHostPorts sends the unique entries in the given hostPorts slice on the
+// returned channel, closing the channel when done. It takes a stop channel,
+// which can be used to signal the sending loop to stop earlier.
+func UniqueHostPorts(stop <-chan struct{}, hostPorts []HostPort) <-chan HostPort {
```

Comment:
:+1: Done - as a side-effect, I've removed DropDuplicatedHostPorts() as it's doing the same thing, only slightly less efficiently for a large input slice.